sui-execution/latest/sui-adapter/src/programmable_transactions/context.rs

```diff
@@ -199,7 +200,8 @@ mod checked {
                 use move_vm_profiler::GasProfiler;
                 use move_vm_types::gas::GasMeter;
 
-                let tx_digest = tx_context.digest();
+                let ref_context: &RefCell<TxContext> = tx_context.borrow();
+                let tx_digest = ref_context.borrow().digest();
```

Comment:
```suggestion
                let tx_digest = tx_context.borrow().digest();
```

Reply:
actually that does not really work for reason I did not quite understand. The rust compiler wanted me to type annotate the `borrow` or to write the code the way I did (which I liked more than the type annotation).
I'll give you the error shortly